coralproject/talk
# Auth refactor

## What does this PR do?
  - Decoupled auth logic into `coral-framework`
    - New reducers
       - `config`
       - `auth`
    - New HOCs
      - `withSignUp`
      - `withSignIn`
      - `withResendEmailConfirmation`
      - `withForgotPassword`
      - `withSetUsername`
   - New Components:
      - `Recaptcha`
  - Decoupled login dialog into its own bundle `coral-login`
  - Implemented admin login on top of new abstractions in `coral-framework`
  - Implemented `talk-plugin-auth` on top of new abstractions in `coral-framework`
  - Decoupled facebook login from `talk-plugin-auth`
  - Implemented client side implementation of facebook login in `talk-plugin-facebook-auth`
  - Fixed Recaptcha bug 
     - Previously you could only send the form once because the recaptcha response is only valid once
     - Now a new recaptcha response is requested for each form send
  - Implemented Recaptcha for the popup login
  - Refactored `admin` and `stream` init into `bootstrap` service
   - Static config is now automatically added to  `config.static`
   - New slots in `talk-plugin-auth`: `authExternalSignIn` and `authExternalSignUp`
   - Exposed new auth abstractions in `plugin-api`

## Note
   - If you have a custom `plugins.json`, `talk-plugin-facebook-auth` must be added to your client side plugins, for it to work.

## Related Stories
- https://www.pivotaltracker.com/story/show/151890174
- https://www.pivotaltracker.com/story/show/154923754

## How do I test this PR?

- Test processes 
  - login
  - register
  - verify e-mail
  - sign up
  - forgot password
  - facebook sign up
  - facebook login
  - login on admin
  - forgot password on admin
  - recaptcha on admin
  - recaptcha on popup login
